scripts/script.js

```diff
@@ -37,48 +41,70 @@ export const options = {
 export default function () {
   // get a random piece from the list
   const piece = pieces[Math.floor(Math.random() * pieces.length)]
-  // run raw vs boost randomly first
-  const runRawFirst = Math.round(Math.random())
+
+  // randomly fetch first from either a raw url or boost
+  const fetchRawUrlFirst = Math.random() >= .5
   let boostResponse, rawResponse
-  for (let i = 0; i < 2; i++) {
-    if ((i + runRawFirst) % 2 === 0) {
-      boostResponse = http.get(`${__ENV.BOOST_FETCH_URL}${piece}`, {
-        tags: {
-          name: 'BoostFetchURL',
-        },
-        timeout: `${__ENV.SIMULTANEOUS_DOWNLOADS}h`,
-      })
-      timeBoost.add(boostResponse.timings.duration)
-      ttfbBoost.add(boostResponse.timings.waiting)
-      boostSuccess.add(
-        boostResponse.status >= 200 && boostResponse.status < 300
-      )
-    } else {
-      if (__ENV.RAW_FETCH_URL) {
-        rawResponse = http.get(`${__ENV.RAW_FETCH_URL}${piece}`, {
-          tags: {
-            name: 'RawFetchURL',
-          },
-          timeout: `${__ENV.SIMULTANEOUS_DOWNLOADS}h`,
-        })
-        timeRaw.add(rawResponse.timings.duration)
-        ttfbRaw.add(rawResponse.timings.waiting)
-        rawSuccess.add(rawResponse.status >= 200 && rawResponse.status < 300)
-      }
-    }
+
+  if (fetchRawUrlFirst) {
+    rawResponse = fetchFromRawUrl(piece)
+    boostResponse = fetchFromBoost(piece)
+  } else {
+    boostResponse = fetchFromBoost(piece)
+    rawResponse = fetchFromRawUrl(piece)
   }
+
   if (__ENV.RAW_FETCH_URL) {
     timeDelta.add(boostResponse.timings.duration - rawResponse.timings.duration)
     ttfbDelta.add(boostResponse.timings.waiting - rawResponse.timings.waiting)
   }
 }
 
-const OUT_DIR = "/out";
+function fetchFromBoost(piece) {
+  let boostResponse = http.get(`${__ENV.BOOST_FETCH_URL}${piece}`, {
+    tags: {
+      name: 'BoostFetchURL',
+    },
+    timeout: `${__ENV.SIMULTANEOUS_DOWNLOADS}h`,
+  })
+  timeBoost.add(boostResponse.timings.duration)
+  ttfbBoost.add(boostResponse.timings.waiting)
+  boostSuccess.add(
+    boostResponse.status >= 200 && boostResponse.status < 300
+  )
+
+  let boostContentLength = parseInt(boostResponse.headers['Content-Length'])
+  dataReceivedBoost.add(boostContentLength, { url: boostResponse.url })
+  bandwidthBoost.add(boostContentLength / boostResponse.timings.duration)
```

Comment:
what is `duration` here, is it milliseconds? seconds?
I'm finding the "bandwidth" numbers not as interesting because I don't know the units, can we build units into the output that explain it?

Reply:
Mmm, yeah, that's probably confusing. Let me see if I can add units to that.

Reply:
So I can add `ms` units to all of the custom time metrics, but I can't say units/time. I'm kind of limited. I think the next best thing is to name the metric `bytes_per_ms_boost`. Thoughts?

Reply:
let's see what @hannahhoward says but I think having the units in the name is a decent solution if you can't annotate it another way, so `bytes_per_ms_X` sgtm

Reply:
I've updated to `bytes_per_ms` for now. The other time units towards the bottom now have `ms` units as well.
![image](https://user-images.githubusercontent.com/3432646/212007734-7aad47cd-8e0a-42ae-bb73-9fee291a8316.png)